rcl_logging_spdlog/src/rcl_logging_spdlog.cpp

```diff
@@ -163,24 +164,27 @@ rcl_logging_ret_t rcl_logging_external_initialize(
     }
     int64_t ms_since_epoch = RCUTILS_NS_TO_MS(now);
 
-    // Get the program name.
-    char * basec = rcutils_get_executable_name(allocator);
-    if (basec == nullptr) {
-      // We couldn't get the program name, so get out of here without setting up
-      // logging.
-      RCUTILS_SET_ERROR_MSG("Failed to get the executable name");
-      return RCL_LOGGING_RET_ERROR;
+    bool file_name_provided = (nullptr != file_name_prefix) && (file_name_prefix[0] != '\0');
+    char * basec;
+    if (file_name_provided) {
+      basec = const_cast<char *>(file_name_prefix);
```

Comment:
I think this would be much better as:

```suggestion
      basec = rcutils_strdup(file_name_prefix, allocator);
```

That way, we can leave the `RCPPUTILS_SCOPE_EXIT` in place below and it will all work properly.  Yes, this is an unnecessary copy, but since this is only called once-per-context, that shouldn't really matter.

Reply:
addressed with https://github.com/ros2/rcl_logging/pull/109/commits/1c6dffc12b6d9342b04cbd072fcf54aab63f918f